discovery-provider/src/tasks/index.py

```diff
@@ -273,89 +297,101 @@ def index_blocks(self, db, blocks_list):
+                    )
+                )
+                user_replica_set_state_changed = total_user_replica_set_changes > 0
+                logger.info(
+                    f"index.py | user_replica_set_state_update completed"
+                    f" user_replica_set_state_changed={user_replica_set_state_changed} for block={block_number}"
+                )
 
-            user_library_state_changed = user_library_state_update( # pylint: disable=W0612
-                self, update_task, session, user_library_factory_txs, block_number, block_timestamp
-            )
-            logger.info(
-                f"index.py | user_library_state_update completed"
-                f" user_library_state_changed={user_library_state_changed} for block={block_number}"
-            )
+                # Playlist state operations processed in bulk
+                total_playlist_changes, playlist_ids = playlist_state_update(
+                    self, update_task, session, playlist_factory_txs, block_number, block_timestamp, block_hash
+                )
+                playlist_state_changed = total_playlist_changes > 0
+                logger.info(
+                    f"index.py | playlist_state_update completed"
+                    f" playlist_state_changed={playlist_state_changed} for block={block_number}"
+                )
 
-            track_lexeme_state_changed = (user_state_changed or track_state_changed)
-            session.commit()
-            logger.info(f"index.py | session commmited to db for block=${block_number}")
-
-            if user_state_changed:
-                if user_ids:
-                    remove_cached_user_ids(redis, user_ids)
-            if user_replica_set_state_changed:
-                if replica_user_ids:
-                    remove_cached_user_ids(redis, replica_user_ids)
-            if track_lexeme_state_changed:
-                if track_ids:
-                    remove_cached_track_ids(redis, track_ids)
-            if playlist_state_changed:
-                if playlist_ids:
-                    remove_cached_playlist_ids(redis, playlist_ids)
-            logger.info(f"index.py | redis cache clean operations complete for block=${block_number}")
+                user_library_state_changed = user_library_state_update( # pylint: disable=W0612
+                    self, update_task, session, user_library_factory_txs, block_number, block_timestamp, block_hash
+                )
+                logger.info(
+                    f"index.py | user_library_state_update completed"
+                    f" user_library_state_changed={user_library_state_changed} for block={block_number}"
+                )
 
+                track_lexeme_state_changed = (user_state_changed or track_state_changed)
+                session.commit()
+                logger.info(f"index.py | session commmited to db for block=${block_number}")
+                if skip_tx_hash:
+                    clear_indexing_error(redis)
```

Comment:
is it possible for a sequence of events to occur where
1. all discprovs hit a tx they can't write
2. some of the discprovs reach consensus that they should skip
3. those some discprovs clear the indexing error
4. other discprovs can't make consensus?

Reply:
The endpoint `/indexing/transaction_status` queries the redis value for indexing error and the DB for skipped transactions - so if some DPs reached consensus and skipped, it would still be exposed and the other DP catching up later will also be able to skip

Reply:
in this case we should probably poll if others have already skipped not if they are currently stuck